lib/active_resource/base.rb

```diff
@@ -1575,20 +1575,17 @@ def find_or_create_resource_for(name)
         resource_name = name.to_s.camelize
 
         const_args = [resource_name, false]
-        if self.class.const_defined?(*const_args)
-          self.class.const_get(*const_args)
-        else
-          ancestors = self.class.name.to_s.split("::")
-          if ancestors.size > 1
-            find_or_create_resource_in_modules(resource_name, ancestors)
-          else
-            if Object.const_defined?(*const_args)
-              Object.const_get(*const_args)
-            else
-              create_resource_for(resource_name)
-            end
-          end
+        return self.class.const_get(*const_args) if self.class.const_defined?(*const_args)
+
+        ancestors = self.class.name.to_s.split("::")
+        return find_or_create_resource_in_modules(resource_name, ancestors) if ancestors.size > 1
+
+        if Object.const_defined?(*const_args)
+          klass = Object.const_get(*const_args)
+          return klass if klass < ActiveResource::Base
         end
+
+        create_resource_for(resource_name)
```

Comment:
> Looks like it boils down to a klass < ActiveResource::Base condition.

That is correct. Please see https://github.com/rails/activeresource/issues/117#issuecomment-300552901 for why I opened this pull request.

> This is intentional: you can represent a resource with a non-ARes class.

I'm not sure why you would want to do that. What is the use case for this?